src/occa/internal/bin/occa.cpp

```diff
@@ -123,92 +135,229 @@ namespace occa {
       return true;
     }
 
+#ifdef BUILD_WITH_OCCA_TRANSPILER
```

Comment:
Let's move all '#ifdef' stuff into new translation unit and expose few functions to abstract which transpiler is used.
For example have:
 bool runTranspiler(...)  
 bool transpileFile(...) 
 int getTranspilerVersion()